andela/ah-frontend-invictus
# #165273442 A user should be able to like dislike articles

#### What does this PR do?
It enables an authenticated user to like and dislike an article 

#### Description of Task to be completed
- Add like dislike presentation component
- Add a container to render the component
- Add actions and types for different cases for liking and disliking
- Add reducers for changing the state fro likes and dislikes
- Add redirection to the login page when the user gets a message of an expired token from the API


#### How should this be manually tested?
Clone the repo and navigate to this branch
Run `npm install` to install necessary dependencies
Run `npm start`
Signup into the platform and login
You will be redirected to the home page
Click `articles` on the navbar
You will view a list of all articles with their likes and dislikes
Click on a particular article to like and or dislike it 

Hit the like button and see the descriptive toast

#### Any background context you want to provide?
Upon liking or disliking the user gets a toast notification message

#### What are the relevant pivotal tracker stories?
[#165273442 Users should be able to like and or dislike articles](https://www.pivotaltracker.com/story/show/165273442)

#### Screenshots (if appropriate)

Display for viewing all articles with their likes
<img width="1359" alt="Screen Shot 2019-06-17 at 10 03 22 AM" src="https://user-images.githubusercontent.com/44635784/59585837-0e68b180-90ea-11e9-9c89-af8274795acd.png">


Message display for successful like

<img width="1395" alt="Screen Shot 2019-06-16 at 9 33 27 PM" src="https://user-images.githubusercontent.com/44635784/59568083-74b7ea80-907e-11e9-8e84-c28959c1f47d.png">


Message display for succesful dislike after like

<img width="1395" alt="Screen Shot 2019-06-16 at 9 34 37 PM" src="https://user-images.githubusercontent.com/44635784/59568093-94e7a980-907e-11e9-9cf2-8a2ae7188d36.png">
